ble/Gap.h

```diff
@@ -419,6 +419,20 @@ class Gap {
      */
 public:
     /**
+     * Returns the current GAP role for the device.
+     */
+    Role_t getRole(void) const {
+        return _role;
+    }
+    
+    /**
+     * Set the GAP role for the device.
+     */
+    void setRole(const Role_t role) {
```

Comment:
who'll call setRole? What happens when a BLE device is concurrently a peripheral and a central?